src/Xamarin.Android.Build.Tasks/Tasks/JavaDependencyVerification.cs

```diff
@@ -349,7 +348,6 @@ public class Package
 public class NuGetPackageVersionFinder
 {
 	readonly LockFile lock_file;
-	readonly Dictionary<string, Artifact> cache = new Dictionary<string, Artifact> ();
 	readonly Regex tag = new Regex ("artifact_versioned=(?<GroupId>.+)?:(?<ArtifactId>.+?):(?<Version>.+)\\s?", RegexOptions.Compiled);
```

Comment:
Sorry I didn't catch this the first time, but For Great Sanity the groups shouldn't try to capture the character following the group, e.g.:

```
(?<GroupId>[^:]+):(?<ArtifactId>[^:]+):(?<Version>.+)\\s?
```

though I wonder if there's a good way to constrain the Version value.

This also applies to `tag2`.

Reply:
Do we have a way to unit test these regex's?  Since they're coming from a NuGet package, it feels like there isn't a good way to unit test things.

Perhaps if we instead had a public `GetArtifactsFromNuspecTags()` method, which we *could* unit test, and have `AddArtifactsFromNuspec()` call `GetArtifactsFromNuspecTags()` passing in `reader.GetTags()`?